### PR TITLE
[MD] UX enhancement for datasource

### PR DESCRIPTION
**Pull request by yibow98**
Signed-off-by: Yibo Wang <yibow@amazon.com>

### Description
This PR is new version of #2501. Previous pr's commit are messy due to merge conflict.
1. Addressed all the comments from previous pr
2. Update snapshots and unit tests accordingly
Screen Shot 2022-10-07 at 12 24 10 AM

 
### Issues Resolved
https://github.com/opensearch-project/ux/issues/19
https://github.com/opensearch-project/ux/issues/23
https://github.com/opensearch-project/ux/issues/24
https://github.com/opensearch-project/ux/issues/25
https://github.com/opensearch-project/ux/issues/26
https://github.com/opensearch-project/ux/issues/29
#2434 
#2439 

 
### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by seraphjiang** [comment]
Thanks @yibow98 , great work
Could we also add @KrooshalUX to reviewers

File: src/core/public/doc_links/doc_links_service.ts [hunk 1]
```diff
@@ -397,6 +397,8 @@ export class DocLinksService {
           functionbeat: `https://opensearch.org/docs/latest/downloads/beats/functionbeat`,
           winlogbeat: `${OPENSEARCH_WEBSITE_DOCS}`,
           siem: `${OPENSEARCH_WEBSITE_DOCS}`,
+          openSearchForum:
+            'https://forum.opensearch.org/t/feedback-experimental-feature-connect-to-external-data-sources/11144',
```
Comment (seraphjiang) [inline src/core/public/doc_links/doc_links_service.ts:401]: try open but get `page not found`?

**Comment by seraphjiang** [comment]
btw, what block us to convert https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2501 from draft to official PR? permission?

File: src/core/public/doc_links/doc_links_service.ts (hunk 1 above)
Comment (yibow98) [reply src/core/public/doc_links/doc_links_service.ts:401]: I asked Kroosh about this link before. The link is correct. It's private until the feature goes live, in order to not drive attention to the forum post.

**Comment by yibow98** [comment]
> btw, what block us to convert #2501 from draft to official PR? permission?

I convert #2501 to draft by myself since the commit history is not correct and will be some problem if that PR merge to Main. That's the reason I create a new PR instead of using previous. Once this PR is merged, I will close that previous PR as soon as possible.

**Comment by mpabba3003** [comment]
@yibow98 #2501 has few same changes that this PR has ..

**Pull request closed without merging**